### PR TITLE
fix(clerk-js): Make Select element themable

### PR DESCRIPTION
**Pull request by desiprisg**
## Type of change

- [x] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend-core`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/edge`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

### CODE REVIEW

File: packages/clerk-js/src/ui/elements/Select.tsx [hunk 1]
```diff
@@ -309,16 +311,13 @@ export const SelectButton = (props: PropsOfComponent<typeof Button>) => {
     <Button
       ref={reference}
       colorScheme='neutral'
-      variant='ghost'
+      variant='select'
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/elements/Select.tsx:314]: ❓ Do we need a variant for that? It looks like we're only using it in a single place. Do you think we can move the corresponding styles inside the Select component directly for now?

Comment (desiprisg) [reply packages/clerk-js/src/ui/elements/Select.tsx:314]: Not right now. I guess we can use the `ghost` variant which uses the same properties from `vars` and override the other properties in the `SelectButton` for now.

File: packages/clerk-js/src/ui/primitives/Button.tsx [hunk 2]
```diff
@@ -8,7 +8,7 @@ import { Spinner } from './Spinner';
 
 const vars = createCssVariables('accent', 'accentDark', 'accentDarker', 'accentLighter', 'accentLightest', 'border');
 
-const { applyVariants, filterProps } = createVariants(theme => {
+const { applyVariants, filterProps } = createVariants((theme, props) => {
```
Comment (nikosdouvlis) [inline packages/clerk-js/src/ui/primitives/Button.tsx:11]: Do we use these props?

**Review by nikosdouvlis — APPROVED** [review]

File: packages/clerk-js/src/ui/primitives/Button.tsx (hunk 2 above)
Comment (desiprisg) [reply packages/clerk-js/src/ui/primitives/Button.tsx:11]: Right, we don't anymore. That got away. Thanks.

**Pull request merged**